upcloud/service/kubernetes.go

```diff
@@ -0,0 +1,127 @@
+package service
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud/request"
+)
+
+type Kubernetes interface {
+	CreateKubernetesCluster(r *request.CreateKubernetesClusterRequest) (*upcloud.KubernetesCluster, error)
+	DeleteKubernetesCluster(r *request.DeleteKubernetesClusterRequest) error
+	GetKubernetesCluster(r *request.GetKubernetesClusterRequest) (*upcloud.KubernetesCluster, error)
+	GetKubernetesClusters(r *request.GetKubernetesClustersRequest) ([]upcloud.KubernetesCluster, error)
+	GetKubernetesClustersWithFilters(r *request.GetKubernetesClustersWithFiltersRequest) ([]upcloud.KubernetesCluster, error)
+	GetKubernetesKubeconfig(r *request.GetKubernetesKubeconfigRequest) (*string, error)
```

Comment:
Maybe return just string instead of pointer ?

Reply:
Fixed.